src/main/java/com/ensao/gi5/lint/rules/AttributeNameRule.java

```diff
@@ -0,0 +1,43 @@
+package com.ensao.gi5.lint.rules;
+
+import com.ensao.gi5.lint.constantes.Constantes;
+import com.ensao.gi5.lint.enumeration.Level;
+import com.ensao.gi5.lint.visitor.AttributeVisitor;
+import com.ensao.gi5.lint.wrapper.AttributeWrapper;
+import com.ensao.gi5.lint.wrapper.CompilationUnitWrapper;
+
+import java.util.ArrayList;
+import java.util.List;
+
+public class AttributeNameRule extends Rule {
+
+    public AttributeNameRule() {
+        super(Constantes.LINT_REG_004, Level.HIGH);
+    }
+    @Override
+    public void apply(CompilationUnitWrapper compilationUnit) {
+        List<AttributeWrapper> attributes = new ArrayList<>();
+        compilationUnit.accept(new AttributeVisitor(), attributes);
+        for (AttributeWrapper attribute: attributes) {
+            if (
+                    Character.isUpperCase(attribute.getFieldName().charAt(0))
+                    && isAttributeNotConstant(attribute)
+            ) {
+                buildViolationThenAddToCollection(
+                        attribute.getLine(),
+                        attribute + " should start with a lower case",
+                        compilationUnit.getFileName()
+                );
+            }
+        }
+    }
+
+    public boolean isAttributeNotConstant(AttributeWrapper attribute) {
+        return !attribute.isStatic() && !attribute.isFinal();
```

Comment:
if it is final, it doesn't mean it is static 
example 
```java 
private final String foo;
```

Reply:
I had to ckeck for constants because they should start with an uppercase thats why i checked for both in this case

Reply:
I see.
a constant is `static && final`,
the opposite is `not (static && final)` so the condition will be `isAttributeNotConstant => not static || not final `

Reply:
I see, thanks for the clarification